schematic/configuration.py

```diff
@@ -0,0 +1,112 @@
+import os
+import yaml
+
+
+class Configuration(object):
+
+    def __init__(self):
+        # this is the project's root directory
+        self.ROOT_DIR = None
+        # path to config.yml file
+        self.CONFIG_PATH = None
+        # path to credentials.json file
+        self.CREDS_PATH = None
+        # path to token.pickle file
+        self.TOKEN_PICKLE = None
+        # path to data directory
+        self.DATA_PATH = None
+        # path to service account credentials file
+        self.SERVICE_ACCT_CREDS = None
+        # path to synapse config file
+        self.SYNAPSE_CONFIG_PATH = None
+        # entire configuration data
+        self.DATA = None
+
+
+    def __getattribute__(self, name):
+        value = super().__getattribute__(name)
+        if value is None and "SCHEMATIC_CONFIG" in os.environ:
+            self.load_config_from_env()
+            value = super().__getattribute__(name)
+        elif value is None and "SCHEMATIC_CONFIG" not in os.environ:
+            raise AttributeError(
+                "The '%s' configuration field was accessed, but it hasn't been "
+                "set yet, presumably because the schematic.CONFIG.load_config() "
+                "method hasn't been run yet. Alternatively, you can re-run this "
+                "code with the 'SCHEMATIC_CONFIG' environment variable set to "
+                "the config.yml file, which will be automatically loaded."
+                % name
+            )
+        return value
+
+
+    def __getitem__(self, key):
+        return self.DATA[key]
+
+
+    @staticmethod
+    def load_yaml(file_path: str) -> dict:
+        with open(file_path, 'r') as stream:
+            try:
+                config_data = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+                return None
+        return config_data
+
+
+    def normalize_path(self, path):
```

Comment:
I thought it might be intuitive to have all non-absolute file paths in the config file to be relative to the config file itself. In most cases, I imagine a folder with the config file and all other associated files (_e.g._ credentials).